server/src-lib/Hasura/GraphQL/Execute/LiveQuery.hs

```diff
@@ -272,39 +271,17 @@ validateAnnVarValsOnPg pgExecCtx annVarVals = do
   let valSel = mkValidationSel $ Map.elems annVarVals
 
   Q.Discard _ <- runTx' $ liftTx $
-    Q.rawQE valPgErrHandler (Q.fromBuilder $ toSQL valSel) [] False
-  return $ fmap (txtEncodedPGVal . snd) annVarVals
+    Q.rawQE defaultTxErrorHandler (Q.fromBuilder $ toSQL valSel) [] False
+  return $ fmap (txtEncodedPGVal . pstValue) annVarVals
 
   where
-    mkExtrs = map (flip S.Extractor Nothing . uncurry toTxtValue)
+    mkExtrs = map (flip S.Extractor Nothing . toTxtValue)
     mkValidationSel vars =
       S.mkSelect { S.selExtr = mkExtrs vars }
     runTx' tx = do
       res <- liftIO $ runExceptT (runLazyTx' pgExecCtx tx)
       liftEither res
 
--- | The error handler that is used to errors in the validation SQL.
--- It tries to specifically read few PG error codes which indicate
--- that the format of the value provided for a type is incorrect
-valPgErrHandler :: Q.PGTxErr -> QErr
```

Comment:
We'll need this exact handler because these are the only error codes that we can expect anything else should be treated as an internal error.